scripts/test-runner.sh

```diff
@@ -116,13 +116,46 @@ for test_suite in "${WRAPPED_TEST_SUITES[@]}"; do
 done
 declare -ri SHOULD_WRAP
 
-if [ "${SHOULD_WRAP}" -eq 0 ]; then
+# This is the list of capabilities to add to the test binary.
+# Note: do not add =e or =i to this setcap command!  We don't want privileged execution by default.
+# Note: if you adjust this list, then you also need to adjust the symmetric list given to the docker run command.
+declare -r CAPS='cap_net_raw,cap_sys_admin,cap_net_admin,cap_sys_rawio=p'
+
+if [ "${TEST_TYPE:-""}" = "FUZZ" ]; then
+  # In this branch we are running full fuzz tests.
+  # These tests are only run from a just command which has already wrapped this script in a docker container.
+
+  # In the case of the full fuzz tests, libstdc++.so.6 will be linked into the test binary because libfuzzer is an LLVM
+  # project (LLVM is a C++ codebase).
+  # Unfortunately, the combination of bolero's RUSTFLAGS and the nix fenix rust overlay _do not_ set the rpath for
+  # libstdc++.so.6.
+  # As a result, a naive attempt to execute the test binary in the compile-env will result in a file not found error
+  # when the dynamic linker is unable to find libstdc++.so.6.
+  # Fortunately, this is relatively easy to fix; we need to patch the test binary to make sure it resolves to the
+  # exact libstdc++.so.6 file which it was liked against.
+  # If the compile-env is correct, then `/lib/libstdc++.so.6` will always be a symlink to the `/nix` store which
+  # contains the correct dynamic library.
+  patchelf --replace-needed libstdc++.so.6 "$(readlink -e /lib/libstdc++.so.6)" "${test_exe}"
```

Comment:
This is really gross.  Is there really no better way with LD_LIBRARY_PATH or editing the rpath in the binary or something?  Perhaps we never care about running this binary elsewhere?

Reply:
Yeah, we don't care about running this binary elsewhere.  It is a one and done sterile build.

I just can't think of any other way to fix this.

LD_LIBRARY_PATH _should_ work, but I can't seem to make it work.  I wish I knew why